src/lib/util/feature-evaluator/client.ts

```diff
@@ -0,0 +1,215 @@
+import { Strategy } from './strategy';
+import { FeatureInterface } from './feature';
+import { RepositoryInterface } from './repository';
+import {
+    Variant,
+    getDefaultVariant,
+    VariantDefinition,
+    selectVariant,
+} from './variant';
+import { Context } from './context';
+import { SegmentForEvaluation } from './strategy/strategy';
+import { PlaygroundStrategySchema } from 'lib/openapi/spec/playground-strategy-schema';
+import { playgroundStrategyEvaluation } from '../../openapi/spec/playground-strategy-schema';
+
+export type StrategyEvaluationResult = Pick<
+    PlaygroundStrategySchema,
+    'result' | 'segments' | 'constraints'
+>;
+
+export type FeatureStrategiesEvaluationResult = {
+    result: boolean | typeof playgroundStrategyEvaluation.unknownResult;
+    strategies: PlaygroundStrategySchema[];
+};
+
+export default class UnleashClient {
```

Comment:
Could it make sense to extend the node SDK itself with the ability to explain its results, instead of duplicating the SDK? Sounds like a nice SDK feature. Not sure how much code has been copied over though.

Reply:
Ah, actually, yeah. Extending the node SDK might be a better way to go. I hadn't thought of that 🙈 Basically, _all_ the code has been copied over, and then I've removed as much as possible. In theory, extending the node SDK instead could be a (glorious) refactor later. That said, there may be a few things that make it harder. It should be fine to override the `isEnabled` functionality as we've done here, but overriding strategy implementations might be harder (and necessary).